### ISSUE TITLE
"make test TESTS=alltests" fail since 3.0.1

### ISSUE THREAD

**Issue by andreasschulze** [openssl/openssl#17366]
It worked in 3.0.0. The failing test is `test_symbol_presence`.
So my workaround is `make tests TEST='alltests -test_symbol_presence'`

The test fail (my opinion only) because I'm using `shlib_variant => "-lcl"`
It looks like [this change](https://github.com/openssl/openssl/commit/73970cb91fdf8e7b4b434d479b875a47a0aa0dbc#diff-877fc978d2f1bf5bbef5a2430df32cde5c1b78a8990dfbe0555ec9b3cb5daac2) break the test.

```
./configdata.pm --dump

Command line (with current working directory = .):

    /usr/bin/perl ./Configure shared --libdir=/usr/local/lib --with-rand-seed=getrandom,os HASHBANGPERL=/usr/bin/perl no-sm2 no-sm3 no-sm4 no-aria enable-ec_nistp_64_gcc_128 no-legacy enable-ktls debian-amd64

Perl information:

    /usr/bin/perl
    5.32.1 for x86_64-linux-gnu-thread-multi

Enabled features:

    afalgeng
    asm
    async
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    bulk
    cached-fetch
    camellia
    capieng
    cast
    chacha
    cmac
    cmp
    cms
    comp
    ct
    deprecated
    des
    dgram
    dh
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ec_nistp_64_gcc_128
    ecdh
    ecdsa
    engine
    err
    filenames
    gost
    idea
    loadereng
    makedepend
    md4
    mdc2
    module
    multiblock
    nextprotoneg
    ocb
    ocsp
    padlockeng
    pic
    pinshared
    poly1305
    posix-io
    psk
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    secure-memory
    seed
    shared
    siphash
    siv
    sock
    srp
    srtp
    sse2
    ssl
    ssl-trace
    static-engine
    stdio
    tests
    threads
    tls
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    acvp-tests          [cascade]              OPENSSL_NO_ACVP_TESTS
    aria                [option]               OPENSSL_NO_ARIA (skip crypto/aria)
    asan                [default]              OPENSSL_NO_ASAN
    buildtest-c++       [default]              
    crypto-mdebug       [default]              OPENSSL_NO_CRYPTO_MDEBUG
    devcryptoeng        [default]              OPENSSL_NO_DEVCRYPTOENG
    egd                 [default]              OPENSSL_NO_EGD
    external-tests      [default]              OPENSSL_NO_EXTERNAL_TESTS
    fips                [default]              
    fips-securitychecks [cascade]              OPENSSL_NO_FIPS_SECURITYCHECKS
    fuzz-afl            [default]              OPENSSL_NO_FUZZ_AFL
    fuzz-libfuzzer      [default]              OPENSSL_NO_FUZZ_LIBFUZZER
    ktls                [not-linux-or-freebsd] OPENSSL_NO_KTLS
    legacy              [option]               
    md2                 [default]              OPENSSL_NO_MD2 (skip crypto/md2)
    msan                [default]              OPENSSL_NO_MSAN
    rc5                 [default]              OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                [default]              OPENSSL_NO_SCTP
    sm2                 [option]               OPENSSL_NO_SM2 (skip crypto/sm2)
    sm3                 [option]               OPENSSL_NO_SM3 (skip crypto/sm3)
    sm4                 [option]               OPENSSL_NO_SM4 (skip crypto/sm4)
    trace               [default]              OPENSSL_NO_TRACE
    ubsan               [default]              OPENSSL_NO_UBSAN
    unit-test           [default]              OPENSSL_NO_UNIT_TEST
    uplink              [no uplink_arch]       OPENSSL_NO_UPLINK
    weak-ssl-ciphers    [default]              OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                [default]              
    zlib-dynamic        [default]              
    ssl3                [default]              OPENSSL_NO_SSL3
    ssl3-method         [default]              OPENSSL_NO_SSL3_METHOD

Config target attributes:

    AR => "ar",
    ARFLAGS => "qc",
    CC => "gcc",
    CFLAGS => "-Wall -O3",
    CXX => "g++",
    CXXFLAGS => "-Wall -O3",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    asm_arch => "x86_64",
    bn_ops => "SIXTY_FOUR_BIT_LONG",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cflags => "-pthread -m64 -Wa,--noexecstack -Wall",
    cppflags => "",
    cxxflags => "-std=c++11 -pthread -m64",
    defines => [ "OPENSSL_BUILDING_OPENSSL" ],
    disable => [  ],
    dso_ldflags => "-Wl,-z,defs",
    dso_scheme => "dlfcn",
    enable => [ "afalgeng" ],
    ex_libs => "-ldl -pthread",
    includes => [  ],
    lflags => "",
    lib_cflags => "",
    lib_cppflags => "-DOPENSSL_USE_NODELETE -DL_ENDIAN",
    lib_defines => [  ],
    module_cflags => "-fPIC",
    module_cxxflags => undef,
    module_ldflags => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    multilib => "64",
    perl_platform => "Unix",
    perlasm_scheme => "elf",
    shared_cflag => "-fPIC",
    shared_defflag => "-Wl,--version-script=",
    shared_defines => [  ],
    shared_ldflag => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    shared_rcflag => "",
    shared_sonameflag => "-Wl,-soname=",
    shared_target => "linux-shared",
    shlib_variant => "-lcl",
    thread_defines => [  ],
    thread_scheme => "pthreads",
    unistd => "<unistd.h>",

Recorded environment:

    AR = 
    ARFLAGS = 
    AS = 
    ASFLAGS = 
    BUILDFILE = 
    CC = 
    CFLAGS = -g -O2 -fdebug-prefix-map=/tmp=. -fstack-protector-strong -Wformat -Werror=format-security
    CPP = 
    CPPDEFINES = 
    CPPFLAGS = -Wdate-time -D_FORTIFY_SOURCE=2
    CPPINCLUDES = 
    CROSS_COMPILE = 
    CXX = 
    CXXFLAGS = -g -O2 -fdebug-prefix-map=/tmp=. -fstack-protector-strong -Wformat -Werror=format-security
    HASHBANGPERL = 
    LD = 
    LDFLAGS = -Wl,-z,relro -Wl,-z,now
    LDLIBS = 
    MT = 
    MTFLAGS = 
    OPENSSL_LOCAL_CONFIG_DIR = 
    PERL = 
    RANLIB = 
    RC = 
    RCFLAGS = 
    RM = 
    WINDRES = 
    __CNF_CFLAGS = 
    __CNF_CPPDEFINES = 
    __CNF_CPPFLAGS = 
    __CNF_CPPINCLUDES = 
    __CNF_CXXFLAGS = 
    __CNF_LDFLAGS = 
    __CNF_LDLIBS = 

Makevars:

    AR              = ar
    ARFLAGS         = qc
    CC              = gcc
    CFLAGS          = -g -O2 -fdebug-prefix-map=/tmp=. -fstack-protector-strong -Wformat -Werror=format-security
    CPPDEFINES      = 
    CPPFLAGS        = -Wdate-time -D_FORTIFY_SOURCE=2
    CPPINCLUDES     = 
    CXX             = g++
    CXXFLAGS        = -g -O2 -fdebug-prefix-map=/tmp=. -fstack-protector-strong -Wformat -Werror=format-security
    HASHBANGPERL    = /usr/bin/perl
    LDFLAGS         = -Wl,-z,relro -Wl,-z,now
    LDLIBS          = 
    PERL            = /usr/bin/perl
    RANLIB          = ranlib
    RC              = windres
    RCFLAGS         = 

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    Makefile

build file:

    Makefile

build file templates:

    Configurations/common0.tmpl
    Configurations/unix-Makefile.tmpl
```

**Comment by mattcaswell** [comment]
Fix in #17372

**Comment by vdukhovni** [comment]
The fix has been merged, should the issue be now closed?
